File: python/selection_sort.py
from base_algorithm import BaseAlgorithm
from isorting_algorithm import ISortingAlgorithm

class SelectionSort(BaseAlgorithm, ISortingAlgorithm):
    def __init__(self):
        super().__init__()
        self.name = "Ordenamiento por Selección"
        self.minimal_complexity = "O(n^2)"
        self.average_complexity = "O(n^2)"
        self.maximal_complexity = "O(n^2)"

    def sort(self, array):
        n = len(array)
        for i in range(n - 1):
            min_index = i
            for j in range(i + 1, n):
                if array[j] < array[min_index]:
                    min_index = j
            if min_index != i:
                array[i], array[min_index] = array[min_index], array[i]

    def get_elemental_operations(self, n):
        return n * n // 2

File: src/sorting/SelectionSort.java
package sorting;

import base.BaseAlgorithm;
import interfaces.ISortingAlgorithm;

public class SelectionSort extends BaseAlgorithm implements ISortingAlgorithm {

    public SelectionSort(){
        this.name = "Ordenamiento por Selección";
        this.minimalComplexity = "O(n^2)";
        this.averageComplexity = "O(n^2)";
        this.maximalCompleity = "O(n^2)";
    }

    @Override
    public void sort(int[] array) {
        int n = array.length;

        for (int i = 0; i < n - 1; i++){
            int minIdex = i;

            for (int j = i; j < n; j++) {
                if (array[j] < array[minIdex]) minIdex = j;
            }
            if (minIdex != i) {
                int temp = array[i];
                array[i] = array[minIdex];
                array[minIdex] = temp;
            }
        }
    }

    @Override
    public Long getElementalOperations(int n) {
        return (long) n * n / 2;
    }
}
